firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java

```diff
@@ -58,21 +59,37 @@ public void addCallback(Consumer<NetworkStatus> callback) {
     callbacks.add(callback);
   }
 
+  @Override
+  public void shutdown() {
+    if (unregisterRunnable != null) {
```

Comment:
This is fine for now, but we should split these two versions so that we're not manually implementing virtual functions just to keep these two different implementations in the same class.